plan/types.go

```diff
@@ -265,8 +265,8 @@ func mergePlanNodes(top, bottom, merged Node) (Node, error) {
 	}
 
 	merged.AddPredecessors(bottom.Predecessors()...)
-	for i, pred := range merged.Predecessors() {
-		for _, succ := range pred.Successors() {
+	for _, pred := range merged.Predecessors() {
+		for i, succ := range pred.Successors() {
```

Comment:
So the bug was caused by this code block using the wrong index? Do I have that right?

Reply:
Yes, that's right. We are looking at the predecessors of the node being replaced, and updating them to point at the new node. But we were using the index from the list of predecessors (which we are not changing) instead of the index from the predecessors' successors.